src/code/z_bgcheck.c

```diff
@@ -4425,27 +4427,24 @@ s32 WaterBox_GetSurfaceImpl(PlayState* play, CollisionContext* colCtx, f32 x, f3
     }
 
     for (i = 0; i < BG_ACTOR_MAX; i++) {
-        if (colCtx->dyna.bgActorFlags[i] & 1) {
-            BgActor* bgActor;
-            if (colCtx->dyna.bgActorFlags[i] & 4) {
-                continue;
-            }
-            bgActor = &colCtx->dyna.bgActors[i];
-            if (bgActor->colHeader->numWaterBoxes != 0 && bgActor->colHeader->waterBoxes != NULL) {
-                for (curWaterBox = colCtx->dyna.waterBoxList.boxes + bgActor->waterboxesStartIndex;
-                     curWaterBox < colCtx->dyna.waterBoxList.boxes + bgActor->waterboxesStartIndex +
-                                       bgActor->colHeader->numWaterBoxes;
-                     curWaterBox++) {
-                    if (curWaterBox->properties & 0x80000) {
-                        continue;
-                    }
-                    if (curWaterBox->minPos.x < x && x < curWaterBox->minPos.x + curWaterBox->xLength) {
-                        if (curWaterBox->minPos.z < z && z < curWaterBox->minPos.z + curWaterBox->zLength) {
-                            *outWaterBox = curWaterBox;
-                            *ySurface = curWaterBox->minPos.y;
-                            *bgId = i;
-                            return true;
-                        }
+        if (!(colCtx->dyna.bgActorFlags[i] & 1) || (colCtx->dyna.bgActorFlags[i] & 4)) {
+            continue;
+        }
+        bgActor = &colCtx->dyna.bgActors[i];
+        if (bgActor->colHeader->numWaterBoxes != 0 && bgActor->colHeader->waterBoxes != NULL) {
+            for (curWaterBox = colCtx->dyna.waterBoxList.boxes + bgActor->waterboxesStartIndex;
+                 curWaterBox <
+                 colCtx->dyna.waterBoxList.boxes + bgActor->waterboxesStartIndex + bgActor->colHeader->numWaterBoxes;
+                 curWaterBox++) {
+                if (curWaterBox->properties & 0x80000) {
+                    continue;
+                }
+                if (curWaterBox->minPos.x < x && x < curWaterBox->minPos.x + curWaterBox->xLength) {
```

Comment:
```suggestion
                if ((curWaterBox->minPos.x < x) && (x < curWaterBox->minPos.x + curWaterBox->xLength)) {
```
Is that the intended way? I don't know if parenthesis can help separate the args to make it clearer

Reply:
It's the way I intended it. I thought I caught them all but apparently not.